release-notes/opensearch.release-notes-2.1.0.md

```diff
@@ -0,0 +1,77 @@
+
+
+#### Deprecations
+
+* [2.x] Deprecate public methods and variables with master term in package 'org.opensearch.action.support.master' (#3617) ([#3643](https://github.com/opensearch-project/opensearch/pull/3643))
+* Deprecate classes in org.opensearch.action.support.master (#3593) ([#3609](https://github.com/opensearch-project/opensearch/pull/3609))
+
+### Security Fixes
+
+* [Dependency upgrade] google-oauth-client to 1.33.3 (#3500) ([#3501](https://github.com/opensearch-project/opensearch/pull/3501))
+
+### Features/Enhancements
+
+* Rename package 'o.o.action.support.master' to 'o.o.action.support.clustermanager' (#3556) ([#3597](https://github.com/opensearch-project/opensearch/pull/3597))
+* Support dynamic node role (#3436) ([#3585](https://github.com/opensearch-project/opensearch/pull/3585))
+* Support use of IRSA for repository-s3 plugin credentials (#3475) ([#3486](https://github.com/opensearch-project/opensearch/pull/3486))
+* Filter out invalid URI and HTTP method in the error message of no handler found for a REST request (#3459) ([#3485](https://github.com/opensearch-project/opensearch/pull/3485))
+* Set term vector flags to false for ._index_prefix field (#1901). (#3119) ([#3447](https://github.com/opensearch-project/opensearch/pull/3447))
+* Replace internal usages of 'master' term in 'server/src/test' directory (#2520) ([#3444](https://github.com/opensearch-project/opensearch/pull/3444))
+* Rename master to cluster_manager in the XContent Parser of ClusterHealthResponse (#3432) ([#3438](https://github.com/opensearch-project/opensearch/pull/3438))
+* Replace internal usages of 'master' term in 'server/src/internalClusterTest' directory (#2521) ([#3407](https://github.com/opensearch-project/opensearch/pull/3407))
+
+
+### Bug Fixes
+
+* [BUG] Custom POM configuration for ZIP publication produces duplicit tags (url, scm) (#3656) ([#3680](https://github.com/opensearch-project/opensearch/pull/3680))
+* [BUG] opensearch crashes on closed client connection before search reply (#3626) ([#3645](https://github.com/opensearch-project/opensearch/pull/3645))
+* Fix false positive query timeouts due to using cached time (#3454) ([#3624](https://github.com/opensearch-project/opensearch/pull/3624))
+* Fix NPE when minBound/maxBound is not set before being called. (#3605) ([#3610](https://github.com/opensearch-project/opensearch/pull/3610))
+* Fix for bug showing incorrect awareness attributes count in AwarenessAllocationDecider (#3428) ([#3580](https://github.com/opensearch-project/opensearch/pull/3580))
+* Add flat_skew setting to node overload decider (#3563) ([#3582](https://github.com/opensearch-project/opensearch/pull/3582))
+* Revert beat types changes 2x  ([#3560](https://github.com/opensearch-project/opensearch/pull/3560))
+* Revert "[Remove] MainResponse version override cluster setting (#3031) (#3032)" ([#3530](https://github.com/opensearch-project/opensearch/pull/3530))
```

Comment:
We should highlight this commit to signify that 2.1 supports version spoofing against 2.0.0 where it was removed. This temporary allows external clients such as beats (broken without this commit; more details in [3484](https://github.com/opensearch-project/OpenSearch/issues/3484)) to work with OpenSearch engine.